### ISSUE TITLE
🐛 Bug Report: Screenreader and keyboard interaction with navigation tabs in pages does not behave according to accessibility requirements

### ISSUE THREAD

**Issue by matthewpennell** [backstage/backstage#18607]
### 📜 Description

When using either the keyboard or a screenreader to interact with plugin navigation tabs, the links are activated (and the associated page content loads) as soon as they receive focus. This is incorrect - according to [Material's own accessibility specification for tabs](https://m3.material.io/components/tabs/accessibility#a1ebaebd-1bb4-4401-9962-8b30a563e49e) the user should be able to move back and forth between navigation items before activating one using Space/Enter.

### 👍 Expected behavior

Keyboard or screenreader users should be able to navigate within the set of links in the navigation bar freely before choosing which link to activate with Space/Enter (or Ctrl+Opt+Space for screenreaders like VoiceOver). The selected state of each tab should not change until it has been activated by the user.

### 👎 Actual Behavior with Screenshots

Screenshots don't help illustrate the problem, happy to provide a screen recording if necessary.

### 👟 Reproduction steps

This was tested on the [Backstage demo site](https://demo.backstage.io/).

1. Navigate to the "Explore" plugin, as this has plugin-level navigation bar.
2. Tab through the page links until you reach the navigation tabs, "Domains" is selected.
3. Press right arrow. Focus shifts to "Groups" tab, which becomes selected, and the contents of that page is loaded.
4. Press right arrow again to activate/load the "Tools" page.

### 📃 Provide the context for the Bug.

The implementation of tabs on Backstage is incorrect according to the accessibility requirements of Material's component, and does not conform to expectations of keyboard or screenreader users. It also performs unnecessary page loads.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by benjdlambert** [comment]
@matthewpennell is this something you'd be willing to help us fix :pray:

**Comment by matthewpennell** [comment]
> @matthewpennell is this something you'd be willing to help us fix 🙏

@benjdlambert Possibly, but my React knowledge is relatively low.

**Comment by JunedKhan101** [comment]
Nevermind, I found it. Let me quickly setup.
